scripts/quests/otherAreas/An_Affable_Adamantking.lua

```diff
@@ -0,0 +1,321 @@
+                    if option == 0 then
+                        if player:getGil() > 100000 then
+                            quest:setVar(player, 'partsWait', getVanaMidnight())
+                            player:delGil(100000)
+                            quest:setMustZone(player)
+                        else
+                            player:messageSpecial(windurstWoodsID.text.NOT_HAVE_ENOUGH_GIL)
+                        end
+                    elseif option == 100 then
+                        player:messageSpecial(zones[player:getZoneID()].text.QUEST_CANCELED)
+                        quest:setVar(player, 'Prog', 0)
+                        player:delQuest(quest.areaId, quest.questId)
+                        player:setVar('BstHeadGearQuest_Conquest', 0) -- player can accept other headgear quests upon cancelling
+                    end
+                end,
+
+                [715] = function(player, csid, option, npc)
+                    if
+                        player:getFreeSlotsCount() > 0 and
+                        not player:hasItem(xi.item.QUADAV_PARTS)
+                    then
+                        if quest:getVar(player, 'Prog') == 2 then
+                            quest:setVar(player, 'Prog', 3) -- only progress if current prog is 2, otherwise player is already at, or beyond, Prog 3
+                        end
+
+                        quest:setVar(player, 'partsWait', 0)
+                        npcUtil.giveItem(player, xi.item.QUADAV_PARTS)
+                    else
+                        player:messageSpecial(windurstWoodsID.text.ITEM_CANNOT_BE_OBTAINED, xi.item.QUADAV_PARTS)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.QULUN_DOME] =
+        {
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    local pos = player:getPos()
+                    if
+                        pos.x >= 0 and
+                        pos.y >= 20 and
+                        pos.z >= 55 and
+                        pos.x <= 6 and
+                        pos.y <= 25 and
+                        pos.z <= 65 and
+                        quest:getVar(player, 'Prog') == 3 and
+                        player:getEquipID(xi.slot.HEAD) == xi.item.QUADAV_BARBUT
+                    then
+                        return 61 -- Give player seeker bats ki to plant
+                    end
+                end,
+            },
+
+            ['Beastmens_Banner'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        player:getEquipID(xi.slot.HEAD) == xi.item.QUADAV_BARBUT and
+                        player:hasKeyItem(xi.ki.ORCISH_SEEKER_BATS)
+                    then
+                        if
+                            GetNPCByID(qulunDomeID.npc.BEASTMENS_BANNER):getLocalVar('engaged') < os.time() and -- if engaged 3min timer has passed
```

Comment:
Since you're checking the same local var twice, it would be more clear to set it in a variable prior to the condition

Reply:
Refactored so engaged is a local variable read once